tests/unittests/test_devmode.py

```diff
@@ -0,0 +1,105 @@
+import datetime
+import json
+import os
+import tempfile
+import unittest
+from unittest.mock import patch
+
+import pytz
+
+from tap_eloqua.client import EloquaClient, strftime
+
+
+class MockResponse:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+
+    def json(self):
+        return self.json_data
+
+
+def mocked_auth_post(*args, **kwargs):
+    return MockResponse({"access_token": "", "refresh_token": "", "expires_in": 5000}, status_code=200)
+
+
+class Test_ClientDevMode(unittest.TestCase):
+    """Test the dev mode functionality."""
+
+    tmpdir = tempfile.mkdtemp()
+    predictable_filename = "eloqua_config.json"
+
+    base_config = {
+        "client_id": "",
+        "client_secret": "",
+        "refresh_token": "",
+        "redirect_uri": "",
+        "user_agent": "",
+    }
+
+    @property
+    def tmp_config_path(self):
+        return os.path.join(self.tmpdir, self.predictable_filename)
+
+    @patch("requests.Session.post", side_effect=mocked_auth_post)
+    def test_client_without_dev_mode(self, mocked_auth_post):
+        """checks if the client can write refresh token and expiry to
+        config."""
+
+        with open(self.tmp_config_path, "w") as ff:
+            json.dump(self.base_config, ff)
```

Comment:
I will strongly suggest deleting the temp config once the unit test is executed. If we don't delete it, I think it will keep on accumulating the temp configs (as the everyday job runs) within circleci server.

Reply:
agreed, but as the cci runner is temporary provisioned and the files created during testing are not preserved.
it will be autodeleted